.github/workflows/publish-next-release.yml

```diff
@@ -0,0 +1,108 @@
+# Description: This workflow runs unit + e2e tests, then publishes UI packages
+#              to `@next-release` NPM tag.
+#
+# Triggered by: merge to `next-release/main`
+
+name: Test and Publish / next-release
+
+concurrency:
+  group: next-release-${{ github.sha }}
+  cancel-in-progress: true
```

Comment:
I don't think we should set up concurrency, since we want every push to the branch be tested.